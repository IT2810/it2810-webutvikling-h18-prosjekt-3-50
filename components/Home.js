import React, { Component } from 'react'

import SessionForm from './components/SessionForm.js'

class Home extends Component {
  static navigationOptions = {
    title: 'Home'
  }

  render() {
    const { navigate } = this.props.navigation

    return (
      <View>

        <SessionForm />
      </View>

    )
  }
}