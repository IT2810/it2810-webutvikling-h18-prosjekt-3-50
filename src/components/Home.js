import React, { Component } from 'react'

import { SafeAreaView, ScrollView } from 'react-native'
import  { View, Fab, Icon} from 'native-base'

import CreateSession from './CreateSession.js'
import CalendarView from './CalendarView.js'
import ShowSession from './ShowSession.js'
import PedometerSensor from './PedometerSensor.js'

class Home extends Component {
  static navigationOptions = {
    title: 'Home'
  }

  render() {
    return (
      <SafeAreaView>
        <ScrollView style={{height: '100%'}}>
          <View style={{paddingBottom: '25%'}}>
            <CalendarView navigation={this.props.navigation}/>
            <PedometerSensor />
            <ShowSession />
          </View>
        </ScrollView>
        <Fab
          testID="addSessionButton"
          onPress={() => this.props.navigation.navigate('CreateSession')}
          style={{ backgroundColor: '#5067FF' }}
          position="bottomRight"
        >
          <Icon name="add" style={{fontSize: 34, lineHeight: 34}} />
        </Fab>
      </SafeAreaView>
    )
  }
}

export default Home
