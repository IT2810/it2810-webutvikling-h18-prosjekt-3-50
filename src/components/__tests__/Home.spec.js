/**
 * @jest-environment jsdom
 */

import React from 'react'
import { shallow } from 'enzyme'
import { Home } from '../Home'
import toJson from 'enzyme-to-json'
import { findByID } from '../../testUtils.js'

describe('Home', () => {
  let wrapper
  let navigateMock

  beforeEach(() => {
    navigateMock = jest.fn()
    const navigation = { navigate: navigateMock }
    wrapper = shallow(<Home navigation={navigation} />)
  })

  it('renders correctly', () => {
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it('navigates when add session button is pressed', () => {
    let addSessionButton = findByID(wrapper, 'addSessionButton')
    addSessionButton.props().onPress()

    expect(navigateMock.mock.calls.length).toBe(1)
  })
})
