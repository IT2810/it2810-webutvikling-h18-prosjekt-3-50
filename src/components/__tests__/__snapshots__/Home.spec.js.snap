// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<_class>
  <ScrollViewMock
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <Styled(ViewNB)
      style={
        Object {
          "paddingBottom": "25%",
        }
      }
    >
      <Connect(CalendarView)
        navigation={
          Object {
            "navigate": [MockFunction],
          }
        }
      />
      <PedometerSensor />
      <Connect(ShowSession) />
    </Styled(ViewNB)>
  </ScrollViewMock>
  <Styled(Fab)
    onPress={[Function]}
    position="bottomRight"
    style={
      Object {
        "backgroundColor": "#5067FF",
      }
    }
    testID="addSessionButton"
  >
    <Styled(Icon)
      name="add"
      style={
        Object {
          "fontSize": 34,
          "lineHeight": 34,
        }
      }
    />
  </Styled(Fab)>
</_class>
`;
